Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponentMode.cpp

```diff
@@ -141,6 +147,44 @@ namespace GradientSignal
         ImageGradientModificationNotificationBus::Handler::BusDisconnect();
     }
 
+    void EditorImageGradientComponentMode::Reflect(AZ::ReflectContext* context)
+    {
+        AzToolsFramework::ComponentModeFramework::ReflectEditorBaseComponentModeDescendant<EditorImageGradientComponentMode>(context);
+    }
+
+    void EditorImageGradientComponentMode::RegisterActions()
+    {
+        AzToolsFramework::PaintBrushManipulator::RegisterActions();
+    }
+
+    void EditorImageGradientComponentMode::BindActionsToModes()
+    {
+        auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(actionManagerInterface, "EditorImageGradientComponentMode - could not get ActionManagerInterface on BindActionsToModes.");
+
+        AZ::SerializeContext* serializeContext = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationRequests::GetSerializeContext);
+
+        AZStd::string modeIdentifier = AZStd::string::format(
+            "o3de.context.mode.%s", serializeContext->FindClassData(azrtti_typeid<EditorImageGradientComponentMode>())->m_name);
+
+        actionManagerInterface->AssignModeToAction(modeIdentifier, "o3de.action.paintBrushManipulator.increaseSize");
+        actionManagerInterface->AssignModeToAction(modeIdentifier, "o3de.action.paintBrushManipulator.decreaseSize");
+        actionManagerInterface->AssignModeToAction(modeIdentifier, "o3de.action.paintBrushManipulator.increaseHardness");
+        actionManagerInterface->AssignModeToAction(modeIdentifier, "o3de.action.paintBrushManipulator.decreaseHardness");
+    }
+
+    void EditorImageGradientComponentMode::BindActionsToMenus()
+    {
+        auto menuManagerInterface = AZ::Interface<AzToolsFramework::MenuManagerInterface>::Get();
+        AZ_Assert(menuManagerInterface, "EditorImageGradientComponentMode - could not get MenuManagerInterface on BindActionsToMenus.");
+
+        menuManagerInterface->AddActionToMenu(EditMenuIdentifier, "o3de.action.paintBrushManipulator.increaseSize", 6000);
+        menuManagerInterface->AddActionToMenu(EditMenuIdentifier, "o3de.action.paintBrushManipulator.decreaseSize", 6001);
```

Comment:
The paintbrush manipulation stuff is used for more than just the Image Gradient. It's also used for the Terrain Macro Material, and _could_ be used by more things in other gems in the future. Seems like the registration for it should be handled somewhere in AzToolsFramework, not tied to the Image Gradient.

Reply:
The main issue that I've got here is that paintbrush consumers shouldn't need to know that the paintbrush has hotkeys for controlling size. If more hotkeys get added in the future for controlling opacity, etc, it should be in one place (the paintbrush), not in every consumer.

The way I was dealing with this from a component mode point of view is that I created methods on the paint brush manipulator to do all the registration, and just had pass-through calls that each component mode needed to make.

Reply:
That is a good point. I can make a helper so that every component mode can just pass their identifier up to the PaintBrushManipulator and the helper takes care of adding the actions to their mode for them.